### PR TITLE
Add support for polynomials of degree 0.

### PR DESCRIPTION
**Pull request by ycopin**
See issue #3574. I did not change the limit to degree 16, but I don't see where it is actually needed.

### CODE REVIEW

**Comment by nden** [comment]
When I wrote them a few years ago I followed numpy's limits on polynomial degree. Certainly they are not needed and can be changed.

**Comment by ycopin** [comment]
I further removed the arbitrary limit degree <= 16.

**Comment by pllim** [comment]
Just for the records:
http://docs.scipy.org/doc/numpy/reference/generated/numpy.polynomial.chebyshev.chebpow.html
```
maxpower : integer, optional
    Maximum power allowed. This is mainly to limit growth of the
    series to unmanageable size. Default is 16
```

File: astropy/modeling/polynomial.py [hunk 1]
```diff
@@ -381,10 +381,11 @@ def fit_deriv(self, x, *params):
         x = np.array(x, dtype=np.float, copy=False, ndmin=1)
         v = np.empty((self.degree + 1,) + x.shape, dtype=x.dtype)
         v[0] = x * 0 + 1
```
Comment (mhvk) [inline astropy/modeling/polynomial.py]: While you are at it, just write
```
v[0] = 1
```
The `x * 0` part must be a leftover where broadcasting was needed or so.

**Comment by nden** [comment]
Ok, so numpy has made it optional now.

File: astropy/modeling/polynomial.py [hunk 2]
```diff
@@ -487,9 +488,10 @@ def fit_deriv(self, x, *params):
...
```
Comment (mhvk) [inline astropy/modeling/polynomial.py]: Same here: just `v[0] = 1`

File: astropy/modeling/polynomial.py [hunk 3]
```diff
@@ -575,9 +577,10 @@ def fit_deriv(self, x, *params):
 
         v = np.empty((self.degree + 1,) + x.shape, dtype=np.float)
         v[0] = x * 0 + 1
```
Comment (mhvk) [inline astropy/modeling/polynomial.py]: And here.

**Comment by embray** [comment]
@nden Thanks for the clarification.  I thought maybe it had something to do with a limitation on one of the fitters, but when I tried to track it down I couldn't find any such thing.  This makes sense.

**Comment by embray** [comment]
@ycopin Do you think you could add a couple tests for this--specifically I want to test evaluating and fitting on polynomials with degree=0, since I can imagine that being a somewhat exceptional case.  If you don't have time I can do it.  Then we just need to add a changelog entry for v1.0.2 and we can fix this.

**Comment by ycopin** [comment]
@embray I can try to add specific tests, but you would have to point me to the file I should modify.

**Comment by embray** [comment]
Putting something in https://github.com/astropy/astropy/blob/master/astropy/modeling/tests/test_polynomial.py would be fine.  Make sure to add docstrings to the tests referencing this issue with a link to the github issue.

**Comment by ycopin** [comment]
Sorry I don't see how to handle that...

**Comment by embray** [comment]
@ycopin If you want you can have a look at http://docs.astropy.org/en/stable/development/testguide.html#writing-tests, otherwise I'll add some tests.

**Comment by charris** [comment]
>Default is 16

It is now 100. I don't know what I was thinking...

**Comment by embray** [comment]
@ycopin Sorry for the delay--I added some tests, and a couple other minor updates to this PR over in ycopin/astropy#1.  If you'll merge it then this PR will be updated to incorporate my changes.

**Comment by embray** [comment]
If you don't have time in the next couple days I'll just open a new PR.

**Comment by ycopin** [comment]
I merged https://github.com/ycopin/astropy/pull/1, but obviously GitHub is still not happy ("This pull request contains merge conflicts that must be resolved."). Let me know if you need anything else from me, and thanks for your help.

**Comment by embray** [comment]
Hmm. Not sure why merge conflicts now but whatever.  You just have to rebase your branch on the latest master now and re-push, per the instructions here:  

http://docs.astropy.org/en/stable/development/workflow/development_workflow.html#rebase

(Or if you already know how to rebase you don't need to follow those exact instructions).

Thanks Yannick

**Comment by ycopin** [comment]
I made a new PR (https://github.com/astropy/astropy/pull/3651) after rebasing.

**Comment by embray** [comment]
@ycopin You don't need to make a new PR.  You just `git push origin <branchname> -f` to update your pull request with the rebased branch.

**Comment by ycopin** [comment]
While still utterly confused by all the git magic, it seems I managed to update the PR wrt latest astropy...

**Comment by mhvk** [comment]
@ycopin - definitely takes a while to get used to the thought behind git. But I at least eventually started to see the logic behind it -- and found it started to make quite a lot of sense.

**Comment by embray** [comment]
@ycopin Sorry, something is still amiss here with this commit 88c9498  I'm not sure what you even did to get that there, but it needn't be.  I'm just going to manually merge this and fix it up, it's no problem.

I found going through the exercises on this page extraordinarily useful to understanding what's going on with git: http://pcottle.github.io/learnGitBranching/

**Comment by embray** [comment]
I think I also made things hard by merging a different PR that had a few conflicts with this one, sorry.

**Comment by embray** [comment]
Merged in d389296db4948ec80d7cde222a4a2a1687ce195e and cherry-picked to the v1.0.x branch in a99616d

**Pull request closed without merging**